app/app.go

```diff
@@ -1024,6 +1025,9 @@ func NewElysApp(
 		app.MasterchefKeeper,
 	)
 
+	// Need to do this otherwise app.PerpetualKeeper.LeverageLpKeeper will be nil
+	app.PerpetualKeeper.LeverageLpKeeper = app.LeveragelpKeeper
```

Comment:
okay, so we can't load leveragelpKeeper before perpetualKeeper ?

Reply:
perpetualKeeper needs leveragelpKeeper
leveragelpKeeper needs perpetualKeeper

Reply:
ah, got it. why do we leveragelp in perpetual ? is there a way to avoid cyclic dependency ? (would be better I think)

Reply:
Sorry, My bad there are 3 levels of cyclic dependencies, masterchefKeeper needs perceptual, perpetual depends on leverage to enable pools, and leverageLp needs masterchef

Reply:
we should absolutely avoid that guys this will cause more problems down the road, similarly to AMM and Perpetual, to avoid mutual dependencies between modules we introduce a new module "accounted pool", and relied on Hooks to avoid cyclic dependencies, can we improve the dependency design here as well @avkr003 cc @amityadav0

Reply:
@cosmic-vagabond I agree with you and I also wanted to avoid this, but the constraint came from the fact only those pools that have been enabled for leverage lp can be enabled for perpetual. Due to time constraint I went ahead with this and improve it later.
My second idea is to automatically enable pools for perpetual which gets enabled for leverage lp through hooks.

Reply:
@avkr003 agreed all those are valid points and solutions worth trying, can we just make sure to track those future changes as part of a new ticket so we make sure we can plan around that as well thank you